docs/scos/dev/front-end-development/202307.0/oryx/building-components/component-implementation.md

```diff
@@ -0,0 +1,209 @@
+
+When a component is server-side rendered, some of the browser APIs are not available. Most commonly known is the `window` or `document` object. Take this into account when implementing custom components.
+
+Oryx renders pages on the server and returns the minimum amount of JavaScript needed. Components doesn't need JavaScript initially, but when a user start interacting with a component, or when the component needs to reflect a certain application state, additional JavaScript needs to be loaded. Loading the component logic at the client side is called _hydration_. Because the component logic is loaded over the network and initialized in the application, hydration is costly. Additionally, the component might need to fetch data from a backend API. Oryx therefore tries to avoid or delay hydration till it is needed.
+
+When developing a component, you need to configure the hydration trigger using the `@hydrate` decorator that can take an event or context. The following example shows how to set up the component to be hydrated when the context is changed:
+
+```ts
+import { resolve } from "@spryker-oryx/di";
+import { ProductMixin } from "@spryker-oryx/product";
+import { hydrate } from "@spryker-oryx/utilities";
+
+@hydrate({ context: ProductContext.SKU })
+export class ProductIdComponent extends ProductMixin(LitElement) {
+  // ...
+}
+```
+
+Alternatively, you can configure hydration to be triggered by a specific event:
+
+```ts
+@hydrate({ event: ["mouseover", "focus"] })
+export class ProductIdComponent extends ProductMixin(LitElement) {
+  // ...
+}
+```
+
+## Component definition
+
+The component implementation you've started building in the first section of this document is not yet registered in the application. To make the component available to the application, you need to register the [component definition](/docs/scos/dev/front-end-development/{{page.version}}/oryx/building-components/component-definition.html).
+
+```ts
+import { componentDef } from "@spryker-oryx/utilities";
+
+export const productIdComponent = componentDef({
+  name: "my-product-id",
+  impl: () => import("./id.component").then((m) => m.ProductIdComponent),
+});
+```
+
+This ensures that whenever the component is used anywhere in the DOM, Oryx lazily loads the associated implementation.
+
+```html
+<div>
+  <my-product-id></my-product-id>
+</div>
+```
+
+## Place the component
+
+After you've implemented and registered the component, you need to use it in the application. For example, place the component on a [page](/docs/scos/dev/front-end-development/{{page.version}}/oryx/building-pages/oryx-pages.html), [composition](/docs/scos/dev/front-end-development/{{page.version}}/oryx/building-pages/oryx-compositions.html), or use it inside  (CMS) content.
+
+Also, you can merge the component into an existing page structure. For example, `before` or `after` an existing component or inside (`prepend` or `append`) the components of an existing composition.
```

Comment:
Why are these in code font?

Reply:
These are the actual values that you can use

Reply:
rephrased